knowledge-base/ChrisCarini/intellij-platform-plugin-verifier-action/action-security.yml

```diff
@@ -0,0 +1,6 @@
+name: 'IntelliJ Platform Plugin Verifier'
+github-token:
+  environment-variable-name: GITHUB_TOKEN
+  permissions:
+    contents: read
+    contents-reason: to avoid github rate limiting #Reference: https://github.com/ChrisCarini/intellij-platform-plugin-verifier-action#github-token-authentication
```

Comment:
As author of this GitHub action, I can confirm this assessment is correct. Thank you @Devils-Knight !

Reply:
Thanks @ChrisCarini for the comment! We have been thinking how to get authors of GitHub Actions more involved in this project. It gets hard to understand the reason for permissions. And there is always a risk that the permissions may change over time, and this repo may get outdated. Any ideas? How did you find this repo?

Reply:
> Thanks @ChrisCarini for the comment!

No problem! :)

> We have been thinking how to get authors of GitHub Actions more involved in this project.

Have you considered trying to proactively reach out to GH Action authors and let them know about your project / company and the benefits they would get from onboarding? Perhaps there is a way to get the "top N" GitHub actions used within workflows (of course, excluding ones already accounted for in this repo) and go from there.

> It gets hard to understand the reason for permissions.

Yeah, I can understand that!

> And there is always a risk that the permissions may change over time, and this repo may get outdated.

Ironic, as this was actually my first thought after I left the comment/approval. "What if I was a bad-actor, secretly, and I got my GitHub action added to this repo, but then decided to change the intent behind the usage of the GH token env-var? That wouldn't be good. How would the 'good-actors' (ie, step-security, in this case) know or get that information propagated."

> Any ideas?

Perhaps excessive, but would it be possible for your application to keep track of the GH actions it is aware of, and monitor for any changes that might add/remove the usage of the `GITHUB_TOKEN` (or other sensitive secret) and then open an issue with the respective information to re-evaluate accordingly?

> How did you find this repo?

I was guided to https://app.stepsecurity.io/ after recently onboarding to OSSF's https://github.com/ossf/scorecard-action GitHub action. From there, I pasted in some of my workflows on my repos (of which some of them use a GitHub action I wrote / maintain myself), and saw that nothing was changed, but the web app told me a ticket was created. Clicking that link lead me to the GH issue, and thus this PR.

Reply:
Thanks @ChrisCarini this is very valuable feedback. Let me make a plan on proactively reaching out to GitHub Action authors.